### PR TITLE
Entity Registry Search

### PR DESCRIPTION
**Pull request by htdvisser**
#### Summary

This PR makes entity search available to non-admin users by considering membership data in the query. It also adds support for searching end devices.

Refs #73 and #1654 

#### Changes

- I started by modifying the membership query so that we can re-use it for search (in addition to regular listing)
- Then I refactored the search queries so that they build on top of the membership query
- Added a small change to leverages Postgres full text search (instead of simple LIKE) when running with Postgres (Cockroach doesn't support FTS)
- Added pagination to search (in both IS and CLI)
- Added search for end devices to IS and CLI.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, database and configuration, according to the stability commitments in `README.md`.
- [x] Testing: The changes are covered with unit tests. The changes are tested manually as well.
- [x] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: cmd/ttn-lw-cli/commands/clients.go [hunk 1]
```diff
@@ -94,17 +94,18 @@ var (
...
 			is, err := api.Dial(ctx, config.IdentityServerGRPCAddress)
 			if err != nil {
 				return err
 			}
-			res, err := ttnpb.NewEntityRegistrySearchClient(is).SearchClients(ctx, req)
+			res, err := ttnpb.NewEntityRegistrySearchClient(is).SearchClients(ctx, req, opt)
 			if err != nil {
 				return err
 			}
+			getTotal()
```
Comment (rvolosatovs) [inline cmd/ttn-lw-cli/commands/clients.go]: This looks very weird - maybe add a comment explaining why return value of this function(which returns a `uint64`) is discarded.

Also/or
```suggestion
_ = getTotal()
```
For additional clarity? Explicit `_` shows makes it obvious this is intentional and not a bug

File: cmd/ttn-lw-cli/commands/end_devices.go [hunk 2]
```diff
@@ -223,6 +223,35 @@ var (
...
+
+			is, err := api.Dial(ctx, config.IdentityServerGRPCAddress)
+			if err != nil {
+				return err
+			}
+			res, err := ttnpb.NewEndDeviceRegistrySearchClient(is).SearchEndDevices(ctx, req, opt)
+			if err != nil {
+				return err
+			}
+			getTotal()
```
Comment (rvolosatovs) [inline cmd/ttn-lw-cli/commands/end_devices.go]: IDEM

File: cmd/ttn-lw-cli/commands/clients.go (hunk 1 above)
Comment (htdvisser) [reply cmd/ttn-lw-cli/commands/clients.go]: This is also how we do it in other places (all list commands), let's please not change all of that in this PR.

Comment (rvolosatovs) [reply cmd/ttn-lw-cli/commands/clients.go]: What does this do exactly?

Comment (htdvisser) [reply cmd/ttn-lw-cli/commands/clients.go]: https://github.com/TheThingsNetwork/lorawan-stack/blob/master/cmd/ttn-lw-cli/commands/pagination.go#L37-L49

**Pull request merged**